aixplain/factories/model_factory.py

```diff
@@ -172,3 +170,182 @@ def get_first_k_assets(
             logging.error(error_message)
             return []
+ 
+    @classmethod
+    def list_host_machines(cls, api_key: Optional[Text] = None) -> List[Dict]:
+        """Lists available hosting machines for model.
+
+        Args:
+            api_key (Text, optional): Team API key. Defaults to None.
+
+        Returns:
+            List[Dict]: List of dictionaries containing information about
+            each hosting machine.
+        """
+        machines_url = urljoin(config.BACKEND_URL, f"sdk/hosting-machines")
+        logging.debug(f"URL: {machines_url}")
+        if api_key:
+            headers = {"x-api-key": f"{api_key}", "Content-Type": "application/json"}
+        else:
+            headers = {"x-api-key": f"{cls.api_key}", "Content-Type": "application/json"}
+        response = _request_with_retry("get", machines_url, headers=headers)
+        response_dicts = json.loads(response.text)
+        for dictionary in response_dicts:
+            del dictionary["id"]
+        return response_dicts
+    
+    @classmethod
+    def list_functions(cls, verbose: Optional[bool] = False, 
+                       api_key: Optional[Text] = None) -> List[Dict]:
+        """Lists supported model functions on platform.
+
+        Args:
+            verbose (Boolean, optional): Set to True if a detailed response 
+                is desired; is otherwise False by default.
+            api_key (Text, optional): Team API key. Defaults to None.
+
+        Returns:
+            List[Dict]: List of dictionaries containing information about
+            each supported function.
+        """
+        functions_url = urljoin(config.BACKEND_URL, f"sdk/functions")
+        logging.debug(f"URL: {functions_url}")
+        if api_key:
+            headers = {"x-api-key": f"{api_key}", "Content-Type": "application/json"}
+        else:
+            headers = {"x-api-key": f"{cls.api_key}", "Content-Type": "application/json"}
+        response = _request_with_retry("get", functions_url, headers=headers)
+        response_dict = json.loads(response.text)
+        if verbose:
+            return response_dict
+        del response_dict["results"]
+        function_list = response_dict["items"]
+        for function_dict in function_list:
+            del function_dict["output"]
+            del function_dict["params"]
+            del function_dict["id"]
+        return response_dict
+    
+    # Will add "always_on" and "is_async" when we support them.
+    # def create_asset_repo(cls, name: Text, hosting_machine: Text, version: Text, 
+    #                       description: Text, function: Text, is_async: bool, 
+    #                       source_language: Text, api_key: Optional[Text] = None) -> Dict:
+    @classmethod
+    def create_asset_repo(cls, name: Text, hosting_machine: Text, version: Text, 
+                          description: Text, function: Text,  source_language: Text,
```

Comment:
@shreyasXplain it is a good suggestion, although since these values come from a CLI we need to add validation and later convert them to the Enums for Function and Languages for model factory member functions.
@mikelam-us-aixplain you may want to convert the string values to Enums ([functions](https://github.com/aixplain/aiXplain/blob/main/aixplain/enums/function.py), [languages](https://github.com/aixplain/aiXplain/blob/main/aixplain/enums/language.py)) in our CLI functions [here](https://github.com/aixplain/aiXplain/pull/58/files#diff-be297f39592e16c4048856282b3cee2bf8b359df6f9df4889402a0d215b79a65R70).

An example usage of Enums is here for [Corpus](https://github.com/aixplain/aiXplain/blob/main/aixplain/factories/corpus_factory.py#L145).